problems/enumerated/problem_10107/problem_10107.py

```diff
@@ -0,0 +1,343 @@
+            self.add(main_group)
+
+        
+    def get_part(self, value, label = None, color = WHITE):
+        seg = Segment(
+            ORIGIN,
+            UNIT * value * RIGHT,
+            label,
+            stroke_width = 6,
+            color = color
+        )
+        return seg
+
+    def time_pass(self, all = True):
+        time = self.time.get_value()
+        
+        daughter_segment = self.daughter_segment
+        daughter_label = self.daughter_label
+        mother_segment = self.mother_segment
+        
+        mother_label = self.mother_label
+
+        x3_daughter_segment = self.x3_daughter_segment
+        
+        daughter_segment.become(self.get_part(
+            7+time,
+            r"{:,}".format(int(time+7.5)),
+            color =GREEN
+        ).align_to(daughter_segment, LEFT+UP))
+        daughter_segment.set_label(MathTex(r"{:,}".format(int(time+7.5))))
+        daughter_label.become(daughter_segment.update_label_pos())
+        
+        mother_segment.become(self.get_part(
+            31+time,
+            r"{:,}".format(int(time+31.5)),
+            color =ORANGE
+        ).align_to(mother_segment, LEFT+UP))
+        mother_segment.set_label(MathTex(r"{:,}".format(int(time+31.5))))
+        mother_label.become(mother_segment.update_label_pos())
+
+        x3_daughter_segment.become(VGroup(
+            daughter_segment.copy(),
+            daughter_segment.copy(),
+            daughter_segment.copy()
+        ).arrange(RIGHT, buff=0).next_to(mother_segment, UP, aligned_edge=LEFT, buff=1))
+
+        brace = self.brace 
+        brace_label = self.brace_label
+
+        brace.become(Brace(x3_daughter_segment, UP))
+        brace_label.become(MathTex(r"3 \cdot", r"{:,}".format(int(time+7.5)), r"=", r"{:,}".format(3*int(time+7.5))).next_to(brace, UP))
+
+        
+        if all:
+            return VGroup(
+                daughter_segment, daughter_label,
+                mother_segment, mother_label,
+                brace, brace_label
+            )
+        else:
+            return VGroup(
+                daughter_segment,
+                mother_segment,
+            )
+
+    def age_difference(self):
+        daughter_label = self.daughter_label
+        mother_label = self.mother_label
+        diff_brece = self.diff_brece
+        sign = MathTex(r"-", r"=", r"24")
+        formula = VGroup(
+            mother_label.copy(),
+            sign[0],
+            daughter_label.copy(),
+            sign[1], sign[2],
+        ).arrange(RIGHT, buff=0.15).next_to(diff_brece, UP)
+
+        self.play(AnimationGroup(
+            ReplacementTransform(mother_label.copy(), formula[0]),
+            Write(formula[1]),
+            ReplacementTransform(daughter_label.copy(), formula[2]),
+            Write(formula[3]), Write(formula[4]), lag_ratio=0.8
+        ), run_time = 1.7)
+        self.play(Indicate(formula[4], 1.1), run_time = 0.8)
+        self.play(FadeOut(formula), run_time = 0.5)
+
+
+
```

Comment:
I wouldn't want to force linting rules, but let's keep several in mind:

- 2 empty lines between file level functions and classes.
- 1 empty line between methods in classes
- no more than 1 empty line between two consecutive code lines (If you want to split better, use comments)
- A single new line at the end of the file (If you put 2 \n-s, you'll have an empty line in the end). You'll  see a `red minus` symbol at the end of the file in github if you forget to put the new line (most editors do it for you, so most probably you won't need to do anything)